src/lesson/lesson01.js

```diff
@@ -262,6 +262,170 @@ Lesson01ExerciseStepPlayer.prototype = {
   },
 };
 
+function Lesson01ExerciseStepPlayerMulti(isExample, nGames, nRowsList,
+                                         nColsList, robotPositionList,
+                                         obstaclePositionsList) {
+  Lesson.LessonStepPlayer.call(this);
+  this.lessonStepPlayers = Array(nGames);
+  for (var i = 0; i < nGames; ++i) {
+    this.lessonStepPlayers[i] = new Lesson01ExerciseStepPlayer(
+      isExample, nRowsList[i], nColsList[i], robotPositionList[i],
+      obstaclePositionsList[i]);
+  }
+  this._solved = !!isExample;
+}
+
+Lesson01ExerciseStepPlayerMulti.prototype = Object.create(
+    Lesson01ExerciseStepPlayer.prototype);
+
+Object.assign(Lesson01ExerciseStepPlayerMulti.prototype, {
+  _initializeElements: function() {
+    nPlayers = this.lessonStepPlayers.length;
+    for (var sp in this.lessonStepPlayers) {
+      stepPlayer = this.lessonStepPlayers[sp];
+      var grid_cell_size = stepPlayer.game.gridCellSize() / nPlayers;
+
+      // Offsets that centralize the canvas.
+      var offset_x = (Constants.RUN_VIEW_SQUARE_DIMENSION / (2 * nPlayers) +
+                      Constants.RUN_VIEW_SQUARE_DIMENSION / nPlayers * sp -
+                      grid_cell_size * stepPlayer.game.n_cols / 2);
+      var offset_y = (Constants.RUN_VIEW_SQUARE_DIMENSION / 2 -
+                      grid_cell_size * stepPlayer.game.n_rows / 2);
+      //this._animator.setOrigin(offset_x, offset_y);
+
+
+      stepPlayer.game.character_position =
+          stepPlayer.game.initial_character_position;
+      var grid = new Animator.SimpleGridElement(
+          'grid' + sp, grid_cell_size, stepPlayer.game.n_rows,
+          grid_cell_size, stepPlayer.game.n_cols);
+      grid.x = offset_x;
+      grid.y = offset_y;
+      this._animator.addElement(grid);
+
+      var character = new ElementFactories.createRobot(
+          'p' + sp, grid_cell_size, grid_cell_size);
+      character.x = offset_x + (0.5 + stepPlayer.game.character_position) *
+          grid_cell_size;
+      character.y = offset_y + (0.5 + stepPlayer.game.n_rows - 1) *
+          grid_cell_size;
+      this._animator.addElement(character);
+
+      for (var i = 1; i < stepPlayer.game.n_rows; ++i) {
+        for (var obstacle_pos in stepPlayer.game.obstacles[i]) {
```

Comment:
It probably works fine in this case, but `for in` is actually a bit non-intuitive sometimes. I think it's better to standardize on simple for loops (http://stackoverflow.com/questions/9329446/for-each-over-an-array-in-javascript)